command/agent/config/config.go

```diff
@@ -83,16 +83,16 @@ type Retry struct {
 
 // Vault contains configuration for connecting to Vault servers
 type Vault struct {
-	Address          string      `hcl:"address"`
-	CACert           string      `hcl:"ca_cert"`
-	CAPath           string      `hcl:"ca_path"`
-	TLSSkipVerify    bool        `hcl:"-"`
-	TLSSkipVerifyRaw interface{} `hcl:"tls_skip_verify"`
-	ClientCert       string      `hcl:"client_cert"`
-	ClientKey        string      `hcl:"client_key"`
-	TLSServerName    string      `hcl:"tls_server_name"`
-	Namespace        string      `hcl:"namespace"`
-	Retry            *Retry      `hcl:"retry"`
+	Address               string      `hcl:"address"`
+	CACert                string      `hcl:"ca_cert"`
+	CAPath                string      `hcl:"ca_path"`
+	TLSSkipVerify         bool        `hcl:"-"`
+	TLSSkipVerifyRaw      interface{} `hcl:"tls_skip_verify"`
+	ClientCert            string      `hcl:"client_cert"`
+	ClientKey             string      `hcl:"client_key"`
+	TLSServerName         string      `hcl:"tls_server_name"`
+	Retry                 *Retry      `hcl:"retry"`
+	LeaseRenewalThreshold float64     `hcl:"lease_renewal_threshold"`
```

Comment:
How is this going to be initialized? Please refer to my previous comment for this part. As is, it is never gonna be set. This Vault config is populated by either whatever is in the environment, or passed in the agent config or the command line.

Reply:
My understanding is this will be initialized when reading the below configuration and this is the only behavior I need for my use case (but can update the PR to allow it to be specified with the environment variables.

I did update the code to allow LeaseRenewalThreshold which [corresponds to the consul-template behavior of using the default value when its nil so if its not specified it will default to .90](https://github.com/hashicorp/consul-template/blob/5fbf7508b0e812dd9766291b17ae9010c804056b/config/vault.go#L382-L384)



```hcl
vault {
  address     = "https://vault.service.hetzner-hil.consul:8200"
  ca_cert     = "/etc/nomad/tls/nomad.ca.pem"
  client_cert = "/etc/nomad/tls/nomad.cert.pem"
  client_key  = "/etc/nomad/tls/nomad.key.pem"

  # Renew secrets at 60% of their lease duration 
  lease_renewal_threshold = 0.60
}

auto_auth {
  method "cert" {
    config = {
      ca_cert     = "/etc/nomad/tls/nomad.ca.pem"
      client_cert = "/etc/nomad/tls/nomad.cert.pem"
      client_key  = "/etc/nomad/tls/nomad.key.pem"
      reload      = true
    }
  }
}

listener "tcp" {
  address = "127.0.0.1:9202"
  tls_disable = true

  profiling {
    unauthenticated_pprof_access = true
    unauthenticated_in_flight_request_access = true
  }
}

cache {
  use_auto_auth_token = true
}

template_config {
  exit_on_retry_failure = true
}

telemetry {
  prometheus_retention_time = "24h"
  disable_hostname = true
  telemetry {
    unauthenticated_metrics_access = true
  }
}

...
template {
  source          = "/etc/vault-agent.d/nomad.cert.pem.tpl"
  destination     = "/etc/nomad/tls/nomad.dummy.consul-template"
  backup          = true
  # ../configure_vault_agent.yml#L8
  command         = "systemctl restart nomad && systemctl reload nomad-vault-agent"
  command_timeout = "95s"
  left_delimiter  = "<%"
  right_delimiter = "%>"

  perms = "0700"
}
```